src/frontend/query_protocol.ml

```diff
@@ -103,6 +103,18 @@ type syntax_doc_result =
     documentation : string 
 }
 
+type ppx_deriver_pos =
+{ 
+  a_start : Lexing.position;
+  a_end : Lexing.position;
+}
+
+type ppx_expand_result = 
+{
+  code : string;
+  deriver : ppx_deriver_pos
```

Comment:
Do you use the `ppx_deriver_pos` anywhere else ? If not you can just inline it. Also this does not only concern `derivers` anymore isn't it ?

Since these are actually the position of the attribute, you could call it `attr_start` / `attr_end` or something like that.